### PR TITLE
Fibonacci Search algorithm added

### PR DESCRIPTION
**Pull request by sanket0x**
Implemented a Fibonacci Search algorithm based on divide and conquer technique

### CODE REVIEW

File: Algorithms/Searching/fibonacci_search.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          cout<<"\nElement not found";
+          return;
+        } else if(item > a[mid])
+            FibonacciSearch(a, mid, end, fib, index-1, item);
+          else
+            FibonacciSearch(a, start, mid, fib, index-2, item);
+}
+
+int main() {
+  int n, i, fib[20], a[10]={3, 7, 55, 86, 7, 15, 26, 30, 46, 95};
```
Comment (Bhavnaharitsa) [inline Algorithms/Searching/fibonacci_search.cpp]: make it dynamic

File: Algorithms/Searching/fibonacci_search.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#include<iostream>
+using namespace std;
+void FibonacciSearch(int *a, int start, int end, int *fib, int index, int item) {
```
Comment (Bhavnaharitsa) [inline Algorithms/Searching/fibonacci_search.cpp]: do we need to use pointers?

File: Algorithms/Stack/balanced_parentheses.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    // Check Empty Stack
+    return (s.empty());
+}
+
+// Driver program to test above function
+int main()
+{
+    string expr = "{()}[]";
```
Comment (Bhavnaharitsa) [inline Algorithms/Stack/balanced_parentheses.cpp]: get it from user

Comment (San-B-09) [reply Algorithms/Stack/balanced_parentheses.cpp]: As we are using recursion, so using pointer will reduce the space complexity.

File: Algorithms/Searching/fibonacci_search.cpp (hunk 2 above)
Comment (sanket0x) [reply Algorithms/Searching/fibonacci_search.cpp]: As we are using recursion, so using pointer will reduce the space complexity.

File: Algorithms/Searching/fibonacci_search.cpp (hunk 1 above)
Comment (sanket0x) [reply Algorithms/Searching/fibonacci_search.cpp]: Made the required changed and made a new pull request

File: Algorithms/Stack/balanced_parentheses.cpp (hunk 3 above)
Comment (sanket0x) [reply Algorithms/Stack/balanced_parentheses.cpp]: resolved the issue by making necessary changes in code

File: Algorithms/Searching/fibonacci_search.cpp [hunk 4]
```diff
@@ -21,14 +21,35 @@ void FibonacciSearch(int *a, int start, int end, int *fib, int index, int item)
...
+{
+    int i;
+    for (i=0; i < size; i++)
+        cout << arr[i] << " ";
+    cout << endl;
+}
+
 int main() {
-  int n, i, fib[20], a[10]={3, 7, 55, 86, 7, 15, 26, 30, 46, 95};
+  int n, i;
```
Comment (Bhavnaharitsa) [inline Algorithms/Searching/fibonacci_search.cpp]: no single letter variables

File: Algorithms/Searching/fibonacci_search.cpp [hunk 5]
```diff
@@ -21,14 +21,35 @@ void FibonacciSearch(int *a, int start, int end, int *fib, int index, int item)
...
+  cout<<"Enter the length of array: ";
+  cin>>n;
+
+  int fib[n], a[n];
```
Comment (Bhavnaharitsa) [inline Algorithms/Searching/fibonacci_search.cpp]: same

File: Algorithms/Searching/fibonacci_search.cpp (hunk 4 above)
Comment (sanket0x) [reply Algorithms/Searching/fibonacci_search.cpp]: Made required changes and pushed back

File: Algorithms/Searching/fibonacci_search.cpp (hunk 5 above)
Comment (sanket0x) [reply Algorithms/Searching/fibonacci_search.cpp]: Made required changes and pushed back

**Comment by sanket0x** [comment]
Added code for implementation of the stack using Linked List

**Comment by Bhavnaharitsa** [comment]
Why 2 different algorithms in same pr?

**Comment by Bhavnaharitsa** [comment]
I meant open 2 different prs not two different commits

**Comment by Bhavnaharitsa** [comment]
This pr is very garbled with improper commit message and irrelevant code changes.
Closing this

**Pull request closed without merging**